tiatoolbox/utils/misc.py

```diff
@@ -370,3 +370,18 @@ def conv_out_size(in_size, kernel_size=1, padding=0, stride=1):
         >>> array([49, 49])
   """
     return (np.floor((in_size - kernel_size + (2 * padding)) / stride) + 1).astype(int)
+
+
+def assert_dtype_int(input_var, message="Input must be integer."):
+    """Generate error if dtype is not int.
+
+    Args:
+        input_var (ndarray): input variable to be tested.
+        message (str): Error message to be displayed.
+
+    Returns:
+        Generates an AssertionError message if input is not an int.
+
+    """
+    if input_var.dtype != int:
+        raise AssertionError(message)
```

Comment:
what will happen if its np.uint8 or any np.uint ? I suggest using the formal form from numpy
```
>>> np.issubdtype(np.array([1,2,3], dtype=np.int32).dtype, np.integer)
True
>>> np.issubdtype(np.array([1,2,3], dtype=np.int8).dtype, np.integer)
True
>>> np.issubdtype(np.array([1,2,3], dtype=np.int64).dtype, np.integer)
True
>>> np.issubdtype(np.array([1,2,3], dtype=np.uint64).dtype, np.integer)
True
>>> np.issubdtype(np.array([1,2,3], dtype=np.float16).dtype, np.integer)
False
```

Reply:
Another point is python `int` may be actually int32 on 32bit system, int atm is int64.

Reply:
Resolved in this commit https://github.com/TIA-Lab/tiatoolbox/commit/2dd48591b0c2526157a54888759ad31324ffb853